controllers/topology/desired_state.go

```diff
@@ -299,6 +327,11 @@ func computeCluster(_ context.Context, s *scope.Scope, infrastructureCluster, co
 
 // computeMachineDeployments computes the desired state of the list of MachineDeployments.
 func computeMachineDeployments(ctx context.Context, s *scope.Scope, desiredControlPlaneState *scope.ControlPlaneState) (scope.MachineDeploymentsStateMap, error) {
+	// Mark all the machine deployments that are currently rolling out.
+	// This captured information will be used for
+	//   - Building the TopologyReconciled condition.
+	//   - Making upgrade decisions on machine deployments.
+	s.UpgradeTracker.MachineDeployments.MarkRollingOut(s.Current.MachineDeployments.RollingOut()...)
```

Comment:
Note: Now that we are marking the currently rolling out machines here we can change our upgrade logic (machine deployment version calculation) a little to rely on this (already using it in `AllowUpgrade` check) and drop the redundant `IsAnyRollingOut` check.   
With this we will get an added benefit of being able to continuously rolling out machine deployments instead of doing in batches. (Makes a difference if we start allowing more than 1 machine deployment upgrade simultaneously.).  

Either way, I dont think we should make any changes for that in this PR. If we choose to do it, it should be a separate PR.

Reply:
agreed.
let's merge this PR and then make the optimization you are suggesting